packages/data/src/plugins/persistence/index.js

```diff
@@ -402,6 +402,124 @@ export function migrateThirdPartyFeaturePreferencesToPreferencesStore(
 	}
 }
 
+/**
+ * Migrates interface 'enableItems' data to the preferences store.
+ *
+ * The interface package stores this data in this format:
+ * ```js
+ * {
+ *     enableItems: {
+ *         singleEnableItems: {
+ * 	           complementaryArea: {
+ *                 'core/edit-post': 'edit-post/document',
+ *                 'core/edit-site': 'edit-site/global-styles',
+ *             }
+ *         },
+ *         multipleEnableItems: {
+ *             pinnedItems: {
+ *                 'core/edit-post': {
+ *                     'plugin-1': true,
+ *                 },
+ *                 'core/edit-site': {
+ *                     'plugin-2': true,
+ *                 },
+ *             },
+ *         }
+ *     }
+ * }
+ * ```
+ * and it should be migrated it to:
+ * ```js
+ * {
+ *     'core/edit-post': {
+ *         complementaryArea: 'edit-post/document',
+ *         pinnedItems: {
+ *             'plugin-1': true,
+ *         },
+ *     },
+ *     'core/edit-site': {
+ *         complementaryArea: 'edit-site/global-styles',
+ *         pinnedItems: {
+ *             'plugin-2': true,
+ *         },
+ *     },
+ * }
+ * ```
+ *
+ * @param {Object} persistence The persistence interface.
+ */
+export function migrateInterfaceEnableItemsToPreferencesStore( persistence ) {
+	const interfaceStoreName = 'core/interface';
+	const preferencesStoreName = 'core/preferences';
+	const state = persistence.get();
+	const sourceEnableItems = state[ interfaceStoreName ]?.enableItems;
+
+	// There's nothing to migrate, exit early.
+	if ( ! sourceEnableItems ) {
+		return;
+	}
+
+	const allPreferences = state[ preferencesStoreName ]?.preferences ?? {};
+
+	// First convert complementaryAreas into the right format.
+	// Use the existing preferences as the accumulator so that the data is
+	// merged.
+	const sourceComplementaryAreas =
+		sourceEnableItems?.singleEnableItems?.complementaryArea;
+	const convertedComplementaryAreas = reduce(
```

Comment:
`Complementary Area` seems like a strange description, is there an opportunity here to rename it to something that better describes what it denotes, eg. `focusedArea`, `openPanel`  - potentially not worth the bother as would require wider code changes, and maybe I am missing something behind the idea of complimentary.

Reply:
There are some docs - https://github.com/WordPress/gutenberg/tree/trunk/packages/interface#complementary-areas.

I agree, I don't think it's the best name. I think the problem is that most people (myself included) won't know what it means for it to be complementary. Even after I've read the docs I have a hard time remembering or understanding.

It might be better if it were just called `area` or `interfaceArea`. Then the action would be `showItemInArea`, which is a lot clearer to me.

There are a few public APIs with this naming now so it's a little hard to roll back on.

Reply:
hmm, yeh, I would need to go back and re-read that description each time as well - but if we were to change it I think that would be a separate PR to this one.